opentelemetry-sdk/src/logs/growable_array.rs

```diff
@@ -0,0 +1,370 @@
+/// The default max capacity for the stack portion of `GrowableArray`.
+const DEFAULT_MAX_STACK_CAPACITY: usize = 10;
+/// The default initial capacity for the vector portion of `GrowableArray`.
+const DEFAULT_INITIAL_VEC_CAPACITY: usize = 5;
+
+#[derive(Debug, Clone, PartialEq)]
+/// A hybrid vector that starts with a fixed-size array and grows dynamically with a vector.
+///
+/// `GrowableArray` uses an internal fixed-size array (`initial`) for storing elements until it reaches
+/// `MAX_STACK_CAPACITY`. When this capacity is exceeded, additional elements are stored in a heap-allocated
+/// vector (`additional`). This structure allows for efficient use of stack memory for small numbers of elements,
+/// while still supporting dynamic growth.
+///
+pub(crate) struct GrowableArray<
+    T: Default + Clone + PartialEq,
+    const MAX_STACK_CAPACITY: usize = DEFAULT_MAX_STACK_CAPACITY,
+    const INITIAL_VEC_CAPACITY: usize = DEFAULT_INITIAL_VEC_CAPACITY,
+> {
+    initial: [T; MAX_STACK_CAPACITY],
```

Comment:
nit: initial -> inline, additional -> overflow could better convey the design.

Reply:
yeah true. updated accordingly.